model/task/db.go

```diff
@@ -1649,16 +1649,17 @@ func AbortTasksForBuild(buildId string, taskIds []string, caller string) error {
 	return err
 }
 
-func AbortTasksForVersion(versionId string, taskIds []string, caller string) error {
+func AbortAndMarkResetTasksForVersion(versionId string, taskIds []string, caller string) error {
 	_, err := UpdateAll(
 		bson.M{
-			VersionKey: versionId,
+			VersionKey: versionId, // Include to improve query.
 			IdKey:      bson.M{"$in": taskIds},
 			StatusKey:  bson.M{"$in": evergreen.TaskAbortableStatuses},
 		},
 		bson.M{"$set": bson.M{
-			AbortedKey:   true,
-			AbortInfoKey: AbortInfo{User: caller},
+			AbortedKey:           true,
+			AbortInfoKey:         AbortInfo{User: caller},
+			ResetWhenFinishedKey: true,
```

Comment:
what does ResetWhenFinishedKey do?

Reply:
This relates to your earlier comment about why I removed SetAbortedTasksResetWhenFinished, that's all that function did. I believe when the task officially aborts we check ResetWhenFinished to determine whether to restart it or just fail it (like when a PR is aborted).